waspc/data/Generator/templates/server/src/auth/utils.ts

```diff
@@ -34,28 +32,88 @@ export const authConfig = {
   successRedirectPath: "{= successRedirectPath =}",
 }
 
-export async function findUserBy(where: Prisma.{= userEntityUpper =}WhereUniqueInput): Promise<{= userEntityUpper =}> {
-  return prisma.{= userEntityLower =}.findUnique({ where });
+export async function findAuthIdentity(providerName: string, providerUserId: string) {
+  return prisma.{= authIdentityEntityLower =}.findUnique({
+    where: {
+      providerName_providerUserId: {
+        providerName,
+        providerUserId: providerUserId.toLowerCase(),
```

Comment:
Aha ok! So the next step is, how likely is that guarantee to stay guaranteed :D?
When code gets maintained, is there some way to ensure both places are sticking to this deal? Maybe we should have some common logic for this? Or if nothing better, comments on both sides explaining "this is done like this because in this othe places we also do that", so however reads this will be aware that there is assumption of this other place doing the same thing and will make sure to preserve that deal.

Reply:
In Haskell we would introduce a special wrapper type that would hold certain guarantees, but here, we need to either wrap it in a "ProviderUserId" object e.g. `{ providerUserId: string }` which is generated with `createProviderUserId(id: number | string): { providerUserId: string }` or ... call the lowercase everywhere.

I'll research this a bit.

Reply:
You could simply have a function `normalizeProviderUserId` that you call in both places. Not perfect, somebody could still easily not use it, but at least it centralizes the logic and makes it clear there is an intention on normalizing it.

Reply:
I've created a concept of `ProviderId` which is an object containing the `providerName` and `providerUserId` which get normalised when we create the `ProviderId` with `createProviderId`. All the functions now expect to receive this object and this way we can have the `providerUserId` normalised each time.